core/cap-0063.md

```diff
@@ -266,6 +351,10 @@ The protocol only defines the application order and it cannot strictly define ho
 - After every thread in the stage has succeeded, the changed ledger entries should be passed to the following stage to be applied in parallel as well
 - While the protocol guarantees that there are no data conflicts between the clusters, it's also possible that clusters have 'sub-clusters' that are independent of each other and thus use a higher degree of parallelism than the protocol allows. This doesn't however have any protocol guarantees in terms of the degree of parallelization.
 
+#### Move fee refund to after the transactions in a stage are applied.
+
+Currently, the fee refund is applied after every transaction. This CAP specifies a change to instead apply refunds after each stage. The meta will also need to be updated. Currently, the soroban fee refund currently exist in `txChangesAfter` under `TransactionMeta`. The refund will stop appearing in this field, and will instead be emitted through `postTxApplyFeeProcessing`, which is in the new `TransactionResultMetaV1` struct, which will be emitted by the new `LedgerCloseMetaV2` struct.
```

Comment:
This is actually incorrect. We should apply the refunds after all transactions have been applied, not after each stage.

Reply:
Are you going to update this then?

Reply:
Ah yeah updated.

Reply:
The CAP still states 'after each stage'

Reply:
Something is going on with github. The [branch](https://github.com/sisuresh/stellar-protocol/commits/63-refund/) has been updated, but I see that the PR is spinning with the message "Recent push is still being processed, and will show up in a bit". I'll try pushing another change.